api-put-object-common.go

```diff
@@ -67,20 +67,41 @@ func isReadAt(reader io.Reader) (ok bool) {
 //  minPartSize - 64MiB
 //  maxMultipartPutObjectSize - 5TiB
 //
-func optimalPartInfo(objectSize int64) (totalPartsCount int, partSize int64, lastPartSize int64, err error) {
+func optimalPartInfo(objectSize int64, configuredPartSize uint64) (totalPartsCount int, partSize int64, lastPartSize int64, err error) {
 	// object size is '-1' set it to 5TiB.
 	if objectSize == -1 {
 		objectSize = maxMultipartPutObjectSize
 	}
+
 	// object size is larger than supported maximum.
 	if objectSize > maxMultipartPutObjectSize {
 		err = ErrEntityTooLarge(objectSize, maxMultipartPutObjectSize, "", "")
 		return
 	}
+
+	if int64(configuredPartSize) > objectSize {
+		err = ErrEntityTooLarge(int64(configuredPartSize), objectSize, "", "")
+		return
+	}
+
+	if objectSize > (int64(configuredPartSize) * maxPartsCount) {
+		err = ErrInvalidArgument("Part size * max_parts(10000) is lesser than input objectSize.")
+	}
+
+	if configuredPartSize < minPartSize {
```

Comment:
@harshavardhana minPartSize is 64mb here.. so this will return an error if the user provides 5Mb as part size

Reply:
Oops it should be 5MiB

Reply:
Fixed